source/hands_on/cql.rst

```diff
@@ -0,0 +1,138 @@
+CQL
+^^^^^^^
+
+Overview
+---------
+Conservative Q-learning (CQL) was first proposed in `Conservative Q-Learning for Offline Reinforcement Learning <https://arxiv.org/abs/2006.04779>`_. Recent advances in reinforcement learning (RL), especially when combined with expressive deep network
+function approximators, have produced promising results. However, applying RL to real-world
+problems consistently poses practical challenges: in contrast to the kinds of data-driven methods
+that have been successful in supervised learning, RL is classically regarded as an active
+learning process, where each training run requires active interaction with the environment. Interaction
+with the real world can be costly and dangerous, and the quantities of data that can be gathered
+online are substantially lower than the offline datasets that are used in supervised learning,
+which only need to be collected once. Offline RL algorithms instead learn from large, previously collected
+datasets, without interaction.
+
+However, directly utilizing existing value-based off-policy RL algorithms in an offline setting generally results
+in poor performance, due to issues with bootstrapping from out-of-distribution actions and overfitting. Thus CQL is proposed as a novel method for learning a conservative Q functions of which the expected value is lower-bounded
```

Comment:
Thus many constrain techniques are added to basic online RL algorithm, cql is one of them which learns ...